tensorflow/compiler/tf2tensorrt/convert/convert_nodes_test.cc

```diff
@@ -1681,20 +1712,149 @@ std::ostream& operator<<(std::ostream& os, const TestParamBase& p) {
+  // tensor, and stores it in the list of inputs (input_data_).
+  //
+  // The TF tensor is always created with concrete static input shape given by
+  // dims. The ITensor can be have static or dynamic shape based on the trt_mode
+  // attribute. The ITensor shape is set automatically according to the trt_mode
+  // parameter, unless the user overrides it with an explicit
+  // partial_input_shape_dims argument.
+  //
+  // Parameters:
+  // - name of the input node
+  // - dims actual dimensions of the tensor that we will use during the test
+  //   (including explicit batch dim)
+  // - values initial values for the TF tensor
+  // - dtype data type of the tensor
+  // - partial_input_shape dimensions which can incude unknown shapes. This can
+  //   be empty, in that case the partial_input_shape will be set automatically
+  //   depending on the trt_mode argument. (This argument also includes explicit
+  //   batch dim).
+  //
+  template <typename T>
+  void AddTestTensor(
+      const string& name, const std::vector<int32>& dims, DataType tf_dtype,
+      const std::vector<T>& values,
+      const std::vector<int32>* partial_input_shape_dims = nullptr) {
+    std::vector<int32> partial_shape;
+    if (partial_input_shape_dims && !partial_input_shape_dims->empty()) {
+      partial_shape = *partial_input_shape_dims;
+    } else {
+      if (trt_mode == TrtTestMode::kDynamicShape) {
+        // In dynamic shape mode we set the all dims unknown.
+        partial_shape = std::vector<int32>(dims.size(), -1);
+      } else {
+        // Use static (known) input shapes.
+        partial_shape = dims;
+      }
+    }
+    AddTestTensorWithExplicitBatchDim(name, partial_shape,
+                                      TfDataTypeToTrt(tf_dtype));
+    if (!values.empty()) {
+      VLOG(2) << "Adding test tensor: " << name << " "
+              << DataTypeString(tf_dtype);
+      InputOutputData data{name, AsTensor(values, dims, tf_dtype)};
+      VLOG(2) << "Added tensor: " << data.name
+              << DataTypeString(data.tensor.dtype());
+      input_data_.push_back(data);
+    }
+  }
+
+  // Adds test tensor (same as above) but with the default tf_dtype defined by
+  // the test params.
+  void AddTestTensor(
+      const string& name, const std::vector<int32>& dims,
+      const std::vector<float>& values = {},
+      const std::vector<int32>* partial_input_shape_dims = nullptr) {
+    AddTestTensor<float>(name, dims, tf_dtype, values,
+                         partial_input_shape_dims);
+  }
+
+  // Builds and runs the converted network. Checks output tensor shape. Tests
+  // output values using a matcher. The network can have multiple input and
+  // output tensors. The inputs are defined by the input_data_ member variable.
+  void BuildAndRun(const string& name,
+                   const std::vector<std::vector<int>>& expected_output_dims,
+                   const Status& conversion_status,
```

Comment:
I think we should remove the expected conversion_status from BuildAndRun and let the call sites to handle the case where conversion is expected to fail.
Shall we rename runtime_status to expected_runtime_status, to be consistent with expected_output_dims?